frontend/src/metabase/setup/components/LicenseTokenStep/LicenseTokenStep.tsx

```diff
@@ -0,0 +1,76 @@
+import { t } from "ttag";
+
+import { color } from "metabase/lib/colors";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import { addUndo } from "metabase/redux/undo";
+import { useStep } from "metabase/setup/useStep";
+import { Button, Divider, Text } from "metabase/ui";
+
+import { submitLicenseToken } from "../../actions";
+import { getIsSetupCompleted } from "../../selectors";
+import { ActiveStep } from "../ActiveStep";
+import { InactiveStep } from "../InactiveStep";
+import type { NumberedStepProps } from "../types";
+
+import { LicenseTokenForm } from "./LicenseTokenForm";
+
+export const LicenseTokenStep = ({ stepLabel }: NumberedStepProps) => {
+  const { isStepActive, isStepCompleted, selectThisStep } =
+    useStep("license_token");
+
+  const storeToken = useSelector(state => state.setup.licenseToken);
+
+  const isSetupCompleted = useSelector(getIsSetupCompleted);
+  const dispatch = useDispatch();
+
+  const handleStepSelect = () => {
+    selectThisStep();
+  };
+
+  const handleValidSubmit = (token: string | null) => {
+    dispatch(
+      addUndo({
+        message: t`Your license is activated`,
+      }),
+    );
+    dispatch(submitLicenseToken(token));
+  };
+
+  const handleNext = () => {
+    dispatch(submitLicenseToken(null));
```

Comment:
can you explain why we need to clear the token when going to the next step? Also, from the PR, it seems that this action is only for analytics (edited: and to determine if we want to go to the next step in reducer).

Reply:
We're not clearing it, the token is local and sent to redux when we submit the step (with a valid token).
If you have a look at `frontend/src/metabase/setup/actions.ts` and the reducer, all but the `welcome` and `language` steps have a thunk (which is 90% only for analytics) and then the reducer changes the state with the answer for the step and updates the step to the following

Reply:
I need to run the code to actually understand what's going on. I think the main point here is that the `Next` button should be renamed to `Skip` to indicate what it actually does. We're not validating the token then click the next button to go to the next step but we can either
1. validate the token and go to the next step when clicking `Activate` button OR
2. Skip activating the token in the first place.

Reply:
I remember I asked Vitorio about "Next" vs "Skip"  but it was before the "Activate"  button navigated to the next step, let me ask him again, you're right with the current implementation "Next" will only be used as "Skip"